enhancements/etcd/etcd-tuning-profiles.md

```diff
@@ -0,0 +1,175 @@
+This enhancement only covers the mvp for a tech preview release of this new feature; a future enhancement will be necessary.
+
+## Motivation
+
+Customers have asked for the ability to change the etcd heartbeat interval and leader election timeout to aid in the stability of their clusters.
+We want to be able to allow for this while minimizing the risk of them setting values that cause issues.
+This will also remove the hardcoded platform-dependent values; certain platforms could have different default profiles to maintain backwards compatibilty with the currently hardcoded values.
+
+### User Stories
+
+* As an adminstrator, I want to change etcd tuning profiles to help increase the stability of my cluster and understand the performance/latency cost of the profile change.
+* As an Openshift support, I want to walk a customer through changing the active etcd profile in a minimal number of steps.
+* As an etcd engineer, I want to easily add and test new profiles and internal profile parameters.
+
+### Goals
+
+* Add the currently known profiles:
+  * Default:
+    - HEARTBEAT_INTERVAL: 100ms
+    - LEADER_ELECTION_TIMEOUT: 1000ms
+  * "Slow" (or other name):
+    - HEARTBEAT_INTERVAL: 500ms
+    - LEADER_ELECTION_TIMEOUT: 2500ms
+* Remove the parameters from the podspec rendering and replace with the Profile.
+* Add a config map to allow an adminstrator to change the profile.
+
+### Non-Goals
+
+* Adding more profiles beyond those listed above.
+* Add an API endpoint to change the profile.
+* Handle changing profile without a workload disruption.
+* Allow users to set arbitrary values for the profile parameters.
+
+## Proposal
+
+The profiles are a layer of abstraction that allow a customer to tweak etcd to run more reliably on their system, while not being so open as to allow them to easily harm their cluster by (knowingly or not) setting bad values for them.
+The values for the two proposed profiles for the Tech Preview of this feature are the current default values (applied to all platform except for) and the values applied for Azure and IBMCloud VPC.
+The latter values have been used successfully in the field for some time so the risk to future cluster is minimal.
+Changing to a "slower" profile will likely incur a performance/latency penalty, but that is likely an acceptable trade for cluster stability.
+
+In this iteration, for the Tech Preview, we will make it clear that changing the profile will require an etcd redeployment and thus a disruption on workloads. In the future enhancement, we can discuss a more seamless transition between profiles.
+We will not allow the user to set arbitrary values for the parameters, they must conform to the profiles values (by way of the profile).
+
+In the Tech Preview, the active profile will be set via a configmap that's applied before the etcd pod starts up. An administrator can update the profile by changing the configmap value and forcing the etcd pods to restart.
```

Comment:
ConfigMaps cannot be used as substitutes for APIs.  Doing so invalidates the code when trying to promote to GA.   We'd have to write the API field and update the code while still in techpreview, so you haven't really saved anything.